src/modules/battery.cpp

```diff
@@ -275,24 +334,36 @@ const std::tuple<uint8_t, float, std::string, float> waybar::modules::Battery::g
     }
 
     return {cap, time_remaining, status, total_power / 1e6};
+#endif
   } catch (const std::exception& e) {
     spdlog::error("Battery: {}", e.what());
     return {0, 0, "Unknown", 0};
   }
 }
 
 const std::string waybar::modules::Battery::getAdapterStatus(uint8_t capacity) const {
+#if defined(__Linux__)
   if (!adapter_.empty()) {
     bool online;
     std::ifstream(adapter_ / "online") >> online;
+#else
+    int state;
+    size_t size_state = sizeof state;
+    if (sysctlbyname("hw.acpi.battery.state", &state, &size_state, NULL,0) != 0) {
+      throw std::runtime_error("sysctl hw.acpi.battery.state failed");
+    }
+    bool online = state == 2;
+#endif
     if (capacity == 100) {
       return "Full";
     }
     if (online) {
       return "Plugged";
     }
     return "Discharging";
+#if defined(__Linux__)
```

Comment:
Better inject `{` at the end of the FreeBSD conditional. Also, order conditionals consistently within the file: either Linux first then FreeBSD or vice versa.

Reply:
Done, thanks.h